xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -353,25 +542,44 @@ public void onConfigChanged(ConfigUpdate update) {
       }
     }
 
-    @Override
-    public void onResourceDoesNotExist(String resourceName) {
-      logger.log(XdsLogLevel.INFO, "Resource {0} is unavailable", resourceName);
-      ConfigOrError parsedServiceConfig =
-          serviceConfigParser.parseServiceConfig(Collections.<String, Object>emptyMap());
-      ResolutionResult result =
-          ResolutionResult.newBuilder()
-              .setServiceConfig(parsedServiceConfig)
-              // let channel take action for no config selector
-              .build();
-      listener.onResult(result);
+    private void cleanUpRdsWatcher() {
+      if (rdsWatcher != null) {
+        logger.log(XdsLogLevel.INFO, "Stop watching RDS resource {0}", rdsResource);
+        xdsClient.cancelRdsResourceWatch(rdsResource, rdsWatcher);
+        rdsResource = null;
+        rdsWatcher = null;
+      }
     }
 
-    @Override
-    public void onError(Status error) {
-      logger.log(
-          XdsLogLevel.WARNING,
-          "Received error from xDS client {0}: {1}", xdsClient, error.getDescription());
-      listener.onError(error);
+    private class RdsResourceWatcherImpl implements RdsResourceWatcher {
+
+      @Override
+      public void onChanged(RdsUpdate update) {
+        if (this != rdsWatcher) {
```

Comment:
Let's not do this now based on assumptions on future XdsClient implementation. It's not clear how race is handled by XdsClient yet.

Reply:
Eliminated. Will reconsider in next steps.